popup.js

```diff
@@ -1,30 +1,44 @@
-var months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
-var days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]
+/**
+ * Over-engineered top-level closure to avoid
+ * anything here from leaking globally.
+ * @param {HTMLElement} timeEl
+ * @param {HTMLElement} dateEl
+ */
+;((timeEl, dateEl) => {
+  ;('use strict')
 
-function clockWrite() {
-	//Set all the time variables
-	var d = new Date(),
-	h = d.getUTCHours(),
-	m = d.getUTCMinutes(),
-	s = d.getUTCSeconds();
-	
-	//Set all the date variables
-	day	= d.getUTCDay(),
-	date = d.getUTCDate(),
-	month = d.getUTCMonth(),
-	year = d.getUTCFullYear();
-	
-	h = (h < 10 ? '0' : '') + h;
-	m = (m < 10 ? '0' : '') + m;
-	s = (s < 10 ? '0' : '') + s;
-	
-	utctime = (h + ':' + m + ':' + s);
-	utcdate = (days[day] + ' ' + months[month] + ' ' + date + ', ' + year);
-	
-	document.getElementById("time").firstChild.nodeValue = utctime;
-	document.getElementById("date").firstChild.nodeValue = utcdate;
-}
+  /** @type {Intl.DateTimeFormatOptions} */
+  const dateFormatOpts = {
+    weekday: 'short',
+    day: 'numeric',
+    month: 'short',
+    year: 'numeric',
+    timeZone: 'UTC',
+  }
 
-document.addEventListener('DOMContentLoaded', function () {
-	setInterval(clockWrite, 1000); clockWrite();
-});
+  /** @type {Intl.DateTimeFormatOptions} */
+  const timeFormatOpts = {
+    hour12: false,
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    timeZone: 'UTC',
+  }
+
+  function clockWrite() {
+    const date = new Date()
+
+    timeEl.firstChild.nodeValue = date.toLocaleTimeString(
+      undefined,
+      timeFormatOpts,
+    )
+    dateEl.firstChild.nodeValue = date.toLocaleDateString(
+      undefined,
+      dateFormatOpts,
+    )
+
+    setTimeout(clockWrite, 1000)
```

Comment:
Instead of setting up a `setInterval` and then calling the function immediately so that it doesn't wait for the whole delay before running initially:

```diff
    function clockWrite() {
      // ...
    }

-   setInterval(clockWrite, 1000)
    clockWrite()
```

we can have the function schedule itself:

```diff
    function clockWrite() {
      // ...
+     setTimeout(clockWrite, 1000)
    }

    clockWrite()
```

Whether or not you view this as an improvement will come down to your opinion of aesthetics and your tolerance for side effects in functions and whether you view `clockWrite` calling itself as an essential step of recipe or an outside effect that's now been shoehorned in. Ultimately, I don't think there's a good reason to do one over the other here, so I've gone with what I find more elegant.